enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala

```diff
@@ -752,4 +752,69 @@ return p""")
     // Then
     result.toList should equal(List(Map("a" -> n)))
   }
+
+  test("Should handle optional match with null parts and distinct without NullPointerException") {
+
+    graph.execute("CREATE (:X {eid: 1})")
+
+    val query =
+      """
+        |  MATCH (x:X)
+        |  WITH x
+        |  OPTIONAL MATCH (x)<-[:L]-(req)
```

Comment:
If the expand needed here, our could we simply `OPTIONAL MATCH (req:NON_EXISTING_LABEL)`?

Reply:
Don't really understand what you meant by this one? @fickludd

Reply:
I just though maybe this query could become even shorter:

```
OPTIONAL MATCH (req:Y)
WITH req
OPTIONAL MATCH (req)<-[:*2]-(y)
RETURN DISTINCT req.eid, y.eid
```